homework-2/validators/userLoginValidator.ts

```diff
@@ -0,0 +1,19 @@
+import { NextFunction, Request, Response } from "express";
+import { UserAttrs } from "../types";
+import UserService from "../services/userService";
+import httpStatus from "http-status";
+import { DUPLICATE_USER_ERROR } from "../config/constants";
+
+export async function userLoginValidator(
+    req: Request<Record<string, unknown>, unknown, UserAttrs>,
```

Comment:
I would add error handling on router level or use error middleware
Please look at https://expressjs.com/ru/guide/error-handling.html

Reply:
Will do it in the scope of the next homework.